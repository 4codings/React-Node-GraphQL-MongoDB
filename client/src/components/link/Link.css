.link {
    display: block;
    color: #000;
    text-decoration: none;
}
