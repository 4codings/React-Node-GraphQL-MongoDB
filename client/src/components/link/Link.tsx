import React, { Component } from 'react';
import PropTypes from 'prop-types';
import history from '../../history';
import * as s from './Link.css';

class Link extends Component {
    static propTypes = {
        to: PropTypes.string.isRequired,
        children: PropTypes.node.isRequired,
    };

    handleClick = (e) => {
        e.preventDefault();
        history.push({
            pathname: e.currentTarget.pathname,
        });
    }

    render() {
        const { to, children, ...otherProps } = this.props;

        return (
            <a href={to} onClick={this.handleClick} className={s.link} {...otherProps}>
                {children}
            </a>
        );
    }
}

export default Link;
