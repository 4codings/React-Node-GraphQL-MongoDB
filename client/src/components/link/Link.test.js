import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Link from './Link.tsx';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer
        .create(<Link to="/">Click here</Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('expects "to" prop to be "/home"', () => {
    const wrapper = mount(<Link to="/home">Click here</Link>);
    expect((wrapper).prop('to')).toEqual('/home');
});

it('expects "children" prop to be "Click here"', () => {
    const wrapper = mount(<Link to="/home">Click here</Link>);
    expect((wrapper).prop('children')).toEqual('Click here');
});

it('renders an anchor element', () => {
    const wrapper = mount(<Link to="/">Click here</Link>);
    expect(wrapper.find('a')).toHaveLength(1);
});

it('calls onClick on click event', () => {
    const wrapper = mount(<Link to="/">Click here</Link>);
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');

    wrapper.instance().forceUpdate();
    wrapper.find('a').simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
});
