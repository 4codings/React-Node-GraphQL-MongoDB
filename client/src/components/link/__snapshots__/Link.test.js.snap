// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="link"
  href="/"
  onClick={[Function]}
>
  Click here
</a>
`;
